### PR TITLE
[Inter mode] av1_encode_mv()

### PR DESCRIPTION
**Pull request by ycho**

### CODE REVIEW

**Comment by ycho** [comment]
Does not harm current behavior of rav1e, but not yet used thus [WIP].
We can still land it though.

File: src/context.rs [hunk 1]
```diff
@@ -2600,3 +2600,173 @@ impl ContextWriter {
...
+}
+#[inline(always)]
+pub fn mv_class_base(mv_class: usize) -> u32 {
+  if mv_class != MV_CLASS_0 {
+    (CLASS0_SIZE << (mv_class as usize + 2)) as u32 }
+  else { 0 }
+}
+#[inline(always)]
+// If n != 0, returns the floor of log base 2 of n. If n == 0, returns 0.
+pub fn log_in_base_2(n: u32) -> u8 {
```
Comment (lu-zero) [inline src/context.rs]: I prepared for you a [generic ilog](https://github.com/xiph/rav1e/pull/448). You may use it :)

File: src/context.rs [hunk 2]
```diff
@@ -2600,3 +2600,173 @@ impl ContextWriter {
     }
   }
 }
+
+/* Symbols for coding which components are zero jointly */
+const MV_JOINTS:usize = 4;
+
+#[derive(Copy, Clone, Debug, PartialEq, PartialOrd)]
+pub enum MV_JOINT_TYPE {
```
[the remark this answers is not in the document]
Comment (ycho) [reply src/context.rs]: Done!

File: src/context.rs (hunk 1 above)
Comment (ycho) [reply src/context.rs]: I will...!

**Pull request merged**